38 document additional knowledge interesting aspects (#39)
cloudogu/dogu-development-docs: docs/important/relevant_functionalities_en.md

Log level control addresses these circumstances.

If a dogu logs too many lines by default and permanently, then administrators or support staff have their lives made harder than necessary.
This is because they have to work through a mountain of lines and separate the logs into relevant and irrelevant information.
In addition, excessive logging (e.g. long stack traces combined with container crash loops) can lead to so much data that the hard disk fills up and log rotation is no longer possible.

#### Log levels and their margins

The log levels are sorted by level of detail from coarse to fine and contain the respective level above. For example, log outputs of the level `WARN` thus also contain log outputs of the level `ERROR`.

This gives the developer a great deal of freedom as to which logs are output at which log level.

To distinguish between relevant and irrelevant, there is of course room for interpretation.
Cloudogu as a Dogu provider supports the customer in making logging as low and as concise as possible for their productive systems.
This means, based on our experience as both software developers and Dogu users, we decide for the customer,

> which log sources, log lines, log levels etc. are appropriate per Dogu and log level.

Reviewer: Simplify? Log-level vs Log-Level?

```
rootLogLevel=$(doguctl config logging/root)
```

#### Extensibility

> If there is a need for additional, Dogu-specific log levels, it is conceivable that the associated keys can also be stored under `/config/<doguname>/logging/`.

Reviewer: This paragraph reads awkward. I'll suggest a sentence which may hopefully be better to grasp.
```suggestion
If there is a need for additional Dogu-specific log levels, the dogu root log level can be supplemented with additional keys (also under `/config/<doguname>/logging/`).
```
